### PR TITLE
fix: close hamburger menu when navigation matches current page (#14191)

### LINKED ISSUE #14191: Hamburger menu item that is not a dropdown acts like a dropdown
**Describe the bug**
When on the hamburger menu, mobile/desktop/etc, the menu item without a 'caret' is acting as a dropdown menu when already on the page. For example, when on the category page for 'camcorder', try to go to 'camcorder' again, and it will act as a dropdown, where we can go back.

**To Reproduce**
1. Using the hamburger menu, go to camcorder
2. open hamburger menu again, and click camcorder again
3. it should not 'open' the camcorder dropdown menu as there's no subcategories for it
4. see video

**Expected behavior**
Should close hamburger menu after clicking camcorder again

**Screenshots/Videos**

https://user-images.githubusercontent.com/29099466/139853237-fb0dcc1f-229b-42ab-ac57-c01dece6b528.MP4

### PR DESCRIPTION
**Pull request by BenjaminEmiliani**
Closes #14191

### CODE REVIEW

File: projects/storefrontlib/cms-components/navigation/navigation/navigation-ui.component.ts [hunk 1]
```diff
@@ -114,11 +114,21 @@ export class NavigationUIComponent implements OnInit, OnDestroy {
     }
   }
 
-  toggleOpen(event: UIEvent): void {
+  toggleOpen(event: UIEvent, navNode: NavigationNode): void {
+    const node = <HTMLElement>event.currentTarget;
+
+    if (typeof navNode.url === 'string') {
+      if (window.location.href.includes(navNode.url) && !navNode.children) {
```
Comment (Pucek9) [inline projects/storefrontlib/cms-components/navigation/navigation/navigation-ui.component.ts]: the second condition seems as not necessary for fix
```suggestion
      if (window.location.href.includes(navNode.url)) {
```

then, I guess you can simplify the fragment:
```ts
    if (
      typeof navNode.url === 'string' &&
      window.location.href.includes(navNode.url)
    ) {
      this.hamburgerMenuService.toggle();
      this.renderer.removeClass(node, 'is-open');

      return;
    }
```

Comment (BenjaminEmiliani) [reply projects/storefrontlib/cms-components/navigation/navigation/navigation-ui.component.ts]: Hey @Pucek9 I tried it with your suggestion but then without the second condition there is bug. To see what I mean, remove the second condition and load the app. Then in mobile view, open the menu, click on 'DIGITAL CAMERAS' then click on 'Shop all Digital Cameras>'. If you try to navigate again to 'DIGITAL CAMERAS', it is not possible and subcategories of Digital Cameras become inaccessible

Comment (Pucek9) [reply projects/storefrontlib/cms-components/navigation/navigation/navigation-ui.component.ts]: @BenjaminEmiliani You're right. So if there are no contraindications, let's simplify only.
```suggestion
      if (typeof navNode.url === 'string' && window.location.href.includes(navNode.url) && !navNode.children) {
```

**Review by Pucek9 — DISMISSED** [review]

**Review by Pucek9 — DISMISSED** [review]

**Comment by BenjaminEmiliani** [comment]
Closing PR, solved by #14422

**Pull request closed without merging**